app/lib/content_item_builder.rb

```diff
@@ -0,0 +1,54 @@
+class ContentItemBuilder
```

Comment:
Could you add a note about the motivation for the `ContentItemBuilder` class? I'm not sure that making testing easier should be the reason we make this change.

Reply:
I think mainly to separate 'holding data' from 'fetching data' and 'caching data'. i think the improved testability is a result of the separation.

Reply:
I think I need to do some reading on the builder class pattern! Aside from testing benefits, I don't see the value in having a builder class here. At the moment there's only one way to get a content item (from the cache/content store call), so there's not much config work to be done.

I think that separating the caching and data model is not really necessary. If the caching logic is complex, we could create a module for cacheable models like we did for registries and include it.

Reply:
So the alternative is we leave it as it is, but then whenever you want to test code involving the content_item model, you have to start stubbing out web request, which adds quite a bit of additional complexity throughout the testing code - and you have to 'know' that that that is how the content_item class gets its data. 

I agree it isn't a massive improvement of the model itself (although I do think it is ever so slightly better) but I do think the testability issue is important.

Reply:
Maybe you could do `ContentItem.for('/services')`.
```ruby
def for(path)
   # get hold of an item, do some fancy stuff.
   new(hash)
end

private
   
   :new
```

Reply:
Ok, did that - also removed all development finder stuff. 👍

Reply:
🍰